spec/acceptance/basic_spec.rb

```diff
@@ -65,8 +65,7 @@
   end
 
   describe curl_command("https://#{host_inventory['fqdn']}/pulp/api/v3/", cacert: "#{certdir}/ca-cert.pem") do
-    its(:response_code) { is_expected.to eq(200) }
-    its(:body) { is_expected.not_to contain('artifacts_list') }
+    its(:response_code) { is_expected.to eq(403) }
```

Comment:
I do wonder if this is correct. Also, why is the HTTP 200 code in the other tests (plugin, content route spec).

Reply:
Doesn't look correct to me :)

Reply:
On the other hand, what it this supposed to be? The api root, listing all the endpoints? Does it work if you pass a cert/key?

Reply:
I *think* it's a side-effect of https://github.com/pulp/pulpcore/pull/1593 (and I've asked in pulp-dev to make sure).

If you want to test an endpoint w/o auth, use `/pulp/api/v3/status/` as we do in line 57.

That said, I am not really sure what this test is supposed to test. It was added as part of https://github.com/theforeman/puppet-pulpcore/commit/a60df532a95dbb3bf951aaf10b85f9b31ba3e635 where we implemented HTTPS handling inside this module, but it isn't saying what it's testing :(

Reply:
The idea was mostly that it showed a difference between authenticated and non-authenticated. With that you can see that certificate authentication works. That's something this module sets up. If you have a better idea of how to test auth, I'm open to suggestions.

Reply:
But that's exactly what it did not test, as access to `/` was explicitly allowed without auth (just the content differed).

I would suggest hitting a *real* API endpoint like `/users` or `/artifacts` or `/tasks`. Without auth you get status=403 and a JSON reply of `{"detail":"Authentication credentials were  provided."}`, with you get status=200 and some actual content.

Reply:
> But that's exactly what it did not test, as access to `/` was explicitly allowed without auth (just the content differed).

I disagree it did not test it. By checking the difference in output I believe I was able to test it. You may be right that it happened to be a side effect. I'd be happy to merge a follow up PR.

Reply:
https://github.com/theforeman/puppet-pulpcore/pull/251